main.c

```diff
@@ -187,14 +214,41 @@ static void output_handle_scale(void *data, struct wl_output *wl_output,
 
 	output->scale = scale;
 }
+static void output_handle_done(void *data, struct wl_output *wl_output) {
+	struct slurp_output *output = data;
+	output->geometry = output->raw_geometry;
```

Comment:
Err, this is quite a lot of complexity. Instead, can we just keep only `geometry` and `logical_geometry`, add the scale as a new field, and divide by the scale in `main` after `output->logical_geometry = output->geometry`?